sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -161,6 +161,7 @@ case class ScriptTransformationExec(
 
               if (scriptOutputReader != null) {
                 if (scriptOutputReader.next(scriptOutputWritable) <= 0) {
+                  proc.waitFor()
```

Comment:
shall we have a timeout here? just in case the process hangs.

Reply:
Yes, it is a good idea. I have added timeout conf in `SQLConf`. Please have a review.